Code/GraphMol/FileParsers/MolSupplier.h

```diff
@@ -296,9 +296,9 @@ class RDKIT_FILEPARSERS_EXPORT SmilesMolSupplier : public MolSupplier {
   void checkForEnd();
 
   bool df_end = false;  // have we reached the end of the file?
-  int d_len = 0;        // total number of smiles in the file
-  int d_next = 0;       // the  molecule we are ready to read
-  int d_line = 0;       // line number we are currently on
+  long d_len = 0;       // total number of smiles in the file
```

Comment:
I expect these should all be size_t or at the least unsigned long

Reply:
> I expect these should all be size_t or at the least unsigned long

That would be logical, but these are set to -1 when a supplier is initialized to indicate that some preprocessing needs to be done.

Reply:
Should we have an API-Change label?